commit b4c17cca15d3079b223be0110ac6cfa49bb68cee
Author: Sage Weil <sweil@redhat.com>
Date:   Wed Sep 19 14:08:49 2018 -0500

    mgr/DaemonServer: allow pg_num increases that abort pending merges
    
    If we are waiting for a PG to merge we can't decrease more, but if we
    were in the process of merging and an increase is requested, we can
    abort the merge by increase pg_num_actual whenever we want.
    
    Signed-off-by: Sage Weil <sage@redhat.com>

diff --git a/src/mgr/DaemonServer.cc b/src/mgr/DaemonServer.cc
index 3fbf791b69e..312e3566394 100644
--- a/src/mgr/DaemonServer.cc
+++ b/src/mgr/DaemonServer.cc
@@ -2266,14 +2266,13 @@ void DaemonServer::adjust_pgs()
 			 << " pg_num " << p.get_pg_num()
 			 << " - still creating initial pgs"
 			 << dendl;
-	      } else if (p.get_pg_num() != p.get_pg_num_pending()) {
+	      } else if (p.get_pg_num() != p.get_pg_num_pending() &&
+			 p.get_pg_num_target() < p.get_pg_num()) {
 		dout(10) << "pool " << i.first
 			 << " target " << p.get_pg_num_target()
 			 << " pg_num " << p.get_pg_num()
-			 << " - pg_num_pending != pg_num, waiting"
+			 << " - decrease and pg_num_pending != pg_num, waiting"
 			 << dendl;
-		// FIXME: we might consider allowing pg_num increases without
-		// waiting for the previously planned merge to complete.
 	      } else if (p.get_pg_num_target() < p.get_pg_num()) {
 		// pg_num decrease (merge)
 		pg_t merge_source(p.get_pg_num() - 1, i.first);
